### PR TITLE
DBZ-4332 Add `row_id` to Oracle event's source info block

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-4332

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/relational/Column.java [hunk 1]
```diff
@@ -155,6 +155,13 @@ default boolean isRequired() {
      */
     String comment();
 
+    /**
+     * Get whether the column maps to an invisible column.
+     *
+     * @return {@code true} if the column is invisible, or {@code false} otherwise
+     */
+    boolean isInvisible();
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/relational/Column.java:163]: I wonder if using the name `invisible` is correct. IMHO `INVISIBLE`  columns in Oracle, MySQL etc. is something completely different and we are processing them. Is not `ROWID` pseudo (synthetic) column, not invisible?

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java [hunk 2]
```diff
@@ -319,6 +320,17 @@ protected Callable<Void> createDataEventsForTableCallable(ChangeEventSourceConte
...
     }
 
+    @Override
+    protected void doPreDispatchCreateDataEventForTable(ResultSet rs, Table table, OracleOffsetContext offsetContext) throws SQLException {
+        try {
+            final String rowId = rs.getString("ROWID");
+            offsetContext.setRowId(rowId);
+        }
+        catch (SQLException e) {
+            e.printStackTrace();
```
Comment (jpechane) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java:330]: Please use `LOGGER`

**Review by jpechane — CHANGES_REQUESTED** [review]
@Naros LGTM, thanks! I left two comments related naming and exception handling. When addressed please feel free to merge youself.

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java (hunk 2 above)
Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java:330]: Oops, that was not intended :)

**Review by jpechane — APPROVED** [review]
@Naros LGTM, thanks! Feel free to merge as soon as CI passes.

**Comment by Naros** [comment]
Looking into the failures, they seem to be related to the changes.

**Comment by Naros** [comment]
@jpechane, after investigating the failures, I've changed to focus only on streaming.

The changes needed to handle snapshots were starting to grow tendrils into the schema history topic and requiring some changes there that are better served as a design discussion before proceeding rather than to try and shoehorn a solution that may not be ideal without discussion.

**Comment by jpechane** [comment]
@Naros `OracleSchemaMigrationIT` failure is unrelated?

**Comment by Naros** [comment]
> @Naros `OracleSchemaMigrationIT` failure is unrelated?

I don't think so, i just reran that class locally against 21-xs and it passed all green, so I'd say it's likely timing:
image

**Comment by jpechane** [comment]
@Naros Applied, thanks!

**Pull request merged**